src/typ/tests.rs

```diff
@@ -226,6 +226,7 @@ fn infer_test() {
         "let tuple(tuple(_, _) as x, _) = tuple(tuple(0, 1.0), []) x",
         "tuple(Int, Float)"
     );
+    assert_infer!("let x: String = \"\" x", "String");
```

Comment:
I'd like to add a test for this snippet, but I'm not exactly sure what the correct behaviour is:

```
let x: tuple(x, x) = tuple(5, 5) x
```

Do you have ideas on whether it should unify or not?